### PR TITLE
add FSM in mqtt

### PR DESCRIPTION
**Pull request by hdy8200**
#### What this PR does / why we need it:
Added instructions for using mutexInstruction in MQTT.

### CODE REVIEW

File: docs/guides/cases/connect-mqtt.md [hunk 1]
```diff
@@ -42,6 +42,11 @@ data:
         protocolPropertyList:
           MQTTTopic: "/test/test2" 
       ... # You can continue to configure commands and corresponding topics according to your own needs, just continue to add according to this format
+  mutexInstructions: | 
+  # Configure mutexInstructions as required. The format is, mutexInstruction:"description"
```
Comment (kris21he) [inline docs/guides/cases/connect-mqtt.md:46]: can we illustration why need to config mutexInstructions here?
If this config is optional, need to emphasis the optional choice than the required words.

File: docs/guides/cases/connect-mqtt.md [hunk 2]
```diff
@@ -42,6 +42,11 @@ data:
...
+      testMutexInstruction1: "Mutex_test1" # change this value
```
Comment (kris21he) [inline docs/guides/cases/connect-mqtt.md:47]: name the mutexInstruction in a meaningful name?
eg, busy topic, free topic?

File: docs/guides/cases/connect-mqtt.md [hunk 3]
```diff
@@ -69,3 +74,25 @@ Return from MQTT ***deviceShifu***:
...
+
+Where `get_topicmsg1` is the embedded query or publish string.
+
+After publishing, query MQTT ***deviceShifu*** again, The returned contents are as follows:
+
+```json
+{"mqtt_message":"test_publish","mqtt_receive_timestamp":"2022-04-29 08:57:59.7397692 +0000 UTC m=+75.407609501"}
+```
+
+Support the publishing of some special messages/instructions(mutexInstruction), mutexInstruction is configured in the yaml file. The corresponding topic in MQTT ***deviceShifu*** will enter the busy state after mutexInstruction is published, and will prevent other messages from being published by ***deviceShifu*** in the busy state. After the mutexInstruction is normally received by the MQTT broker, the corresponding topic returns to the idle state.
```
Comment (kris21he) [inline docs/guides/cases/connect-mqtt.md:94]: describe a detailed use case of how to use this function.
eg. 1. describe a scenario that need the mutexInstruction.
       2. how to config
       3. how to use
       4. what is the result of this usage

File: docs/references/api/deviceshifu-configmap.md [hunk 4]
```diff
@@ -65,6 +65,11 @@ instructions: |
         protocolPropertyList:
           MQTTTopic: "/test/test2"
       ...  # You can continue to configure commands and corresponding topics according to your own needs, just continue to add according to this format
+mutexInstructions: | 
+    # Configure mutexInstructions as required. The format is, mutexInstruction:"description"
```
Comment (kris21he) [inline docs/references/api/deviceshifu-configmap.md:69]: shall we illustration the format as key, value? and tell them what does this config is used for, and what the keys stands for, what value they should put? is there any validation of the key and value?

File: docs/tutorials/demo-try.md [hunk 5]
```diff
@@ -466,6 +466,13 @@ sudo kubectl exec -it nginx -- curl http://deviceshifu-mqtt.deviceshifu.svc.clus
 ![deviceshifu-mqtt_output4.png](images/deviceshifu-mqtt_output4.png)
 ![deviceshifu-mqtt_output5.png](images/deviceshifu-mqtt_output5.png)
 
+We can also send command to publish data to multiple topics through the digital twin of MQTT.
```
Comment (kris21he) [inline docs/tutorials/demo-try.md:469]: change to -> "example of using multiple topics in deviceshifu-mqtt:" ?

File: docs/references/advanced-features/fsm.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+Implement the simplest FSM model in MQTT ***DeviceShifu***. The initial state of the device is idle. After using MQTT ***DeviceShifu*** to send instructions to the device, the device transfers from the idle state to the busy state (when the device is in the busy state, other instructions are blocked from receiving); When the device completes the instruction, the device returns the completion information to the MQTT Broker, and the device transfers from the busy state to the idle state.
+
+![images/FSM-MQTT-output2.png](images/FSM-MQTT-output2.png)
+
+![images/FSM-MQTT-output1.png](images/FSM-MQTT-output1.png)
+
+### Functionality
+
+Run the following command in your cluster to send `moving_the_device` command to the device, and the device enters the busy state:
```
Comment (tomqin93) [inline docs/references/advanced-features/fsm.md:25]: which topic are we sending this to?

File: docs/guides/cases/connect-mqtt.md [hunk 7]
```diff
@@ -42,6 +42,11 @@ data:
         protocolPropertyList:
           MQTTTopic: "/test/test2" 
       ... # You can continue to configure commands and corresponding topics according to your own needs, just continue to add according to this format
+  mutexInstructions: | 
+  # Optional, configure mutexInstructions. the format as key, value. The key is the mutex instruction sent to the device. When the device executes the mutex instruction, it enters the busy state and refuses to receive other instructions. The value is the response returned by the device to the MQTT broker, indicating that the device has completed the corresponding mutex instruction and recovered to the idle state
+    Moving_the_device: "Device_finished_moving" # change this row
+    Rotating_the_device: "Device_finished_Rotating"
```
Comment (tomqin93) [inline docs/guides/cases/connect-mqtt.md:48]: How is the `mutexInstructions` link to the above `instructions`?

**Review by tomqin93 — COMMENTED** [review]
@hdy8200 added some comment,please take a look

**Review by tomqin93 — APPROVED** [review]

**Pull request merged**